lib/api/src/modules/stories.ts

```diff
@@ -222,94 +223,96 @@ const initStoriesApi = ({
       kind.match(/\.|\|/)
     );
 
-    const storiesHashOutOfOrder = Object.values(input).reduce((acc, item) => {
-      const { kind, parameters } = item;
-      // FIXME: figure out why parameters is missing when used with react-native-server
-      const {
-        hierarchyRootSeparator: rootSeparator = undefined,
-        hierarchySeparator: groupSeparator = undefined,
-        showRoots = undefined,
-      } = (parameters && parameters.options) || {};
-
-      const usingShowRoots = typeof showRoots !== 'undefined';
-
-      // Kind splitting behaviour as per https://github.com/storybookjs/storybook/issues/8793
-      let root = '';
-      let groups: string[];
-      // 1. If the user has passed separators, use the old behaviour but warn them
-      if (typeof rootSeparator !== 'undefined' || typeof groupSeparator !== 'undefined') {
-        warnRemovingHierarchySeparators();
-        if (usingShowRoots) warnUsingHierarchySeparatorsAndShowRoots();
-        ({ root, groups } = parseKind(kind, {
-          rootSeparator: rootSeparator || '|',
-          groupSeparator: groupSeparator || /\/|\./,
-        }));
-
-        // 2. If the user hasn't passed separators, but is using | or . in kinds, use the old behaviour but warn
-      } else if (anyKindMatchesOldHierarchySeparators && !usingShowRoots) {
-        warnChangingDefaultHierarchySeparators();
-        ({ root, groups } = parseKind(kind, { rootSeparator: '|', groupSeparator: /\/|\./ }));
-
-        // 3. If the user passes showRoots, or doesn't match above, do a simpler splitting.
-      } else {
-        const parts: string[] = kind.split('/');
-        if (showRoots && parts.length > 1) {
-          [root, ...groups] = parts;
+    const storiesHashOutOfOrder = Object.entries(input)
+      .map(([k, v]) => v)
+      .reduce((acc, item) => {
+        const { kind, parameters } = item;
+        // FIXME: figure out why parameters is missing when used with react-native-server
+        const {
+          hierarchyRootSeparator: rootSeparator = undefined,
+          hierarchySeparator: groupSeparator = undefined,
+          showRoots = undefined,
+        } = { ...provider.getConfig(), ...((parameters && parameters.options) || {}) };
+
+        const usingShowRoots = typeof showRoots !== 'undefined';
+
+        // Kind splitting behaviour as per https://github.com/storybookjs/storybook/issues/8793
+        let root = '';
+        let groups: string[];
+        // 1. If the user has passed separators, use the old behaviour but warn them
+        if (typeof rootSeparator !== 'undefined' || typeof groupSeparator !== 'undefined') {
+          warnRemovingHierarchySeparators();
+          if (usingShowRoots) warnUsingHierarchySeparatorsAndShowRoots();
+          ({ root, groups } = parseKind(kind, {
+            rootSeparator: rootSeparator || '|',
```

Comment:
Are we removing these in 6.0? Or are we planning to patch this back to 5.3? Or will we do that in a separate PR?

Reply:
Yeah removing those deprecations is on my wishlist

Reply:
I know a single key on your keyboard that could make your wish come true 😉

Reply:
I'm well aware of existence of the `delete` key on my keyboard

Reply:
![Screenshot 2020-02-13 at 14 59 27](https://user-images.githubusercontent.com/3070389/74442300-8d78da00-4e71-11ea-98f4-2f8ec01ea6d2.png)